From b3d41cd329d1df47405aaa5d7094cfded6a20c85 Mon Sep 17 00:00:00 2001
From: Vladimir Zhbanov <vzhbanov@gmail.com>
Date: Thu, 30 Apr 2015 15:44:23 +0300
Subject: [PATCH 8/9] gaf: Build without readline support under MinGW.

Prohibit activating the readline library as it is not supported for
MinGW build yet, so invoking 'gaf shell' yields the following error:
  [scm-error misc-error #f "~A ~S" ("no code for module" (ice-9 readline))
---
 gaf/shell.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/gaf/shell.c b/gaf/shell.c
index fc08d08..043da49 100644
--- a/gaf/shell.c
+++ b/gaf/shell.c
@@ -116,12 +116,15 @@ cmd_shell_impl (void *data, int argc, char **argv)
    * version message. */
   if (interactive) {
     fprintf (stderr, "gEDA %s (g%.7s)\n", PACKAGE_DOTTED_VERSION, PACKAGE_GIT_COMMIT);
+  /* readline is not supported for MinGW builds yet */
+#ifndef __MINGW32__
     SCM expr = scm_list_3 (sym_begin,
                            scm_list_2 (sym_use_modules,
                                        scm_list_2 (sym_ice_9, sym_readline)),
                            scm_list_1 (sym_activate_readline));
 
     scm_eval_x (expr, scm_current_module ());
+#endif /* __MINGW32__ */
   }
 
   /* Now load rc files, if necessary */
-- 
2.1.4

